data/calculated.py

```diff
@@ -180,4 +180,15 @@ def isel(self, **kwargs):
         for d in self.dims:
             key.append(keys[d])
 
-        return self[tuple(key)]
+        selection = self[tuple(key)]
+        # check to see that dimensions are named, if not recreate DataArray w/ correct dims and coords
+        if selection.dims != tuple(keys.keys()):   
+            shape = [k.stop - k.start for k in keys.values()]
+            selection = xr.DataArray(
+                            data = np.reshape(selection.data, shape), 
+                            dims = keys.keys(),
+                            coords = {str(k) : self._parent.coords[k][v] for k,v in keys.items()},
+                            attrs = self.attrs
+                        )
```

Comment:
Since we're already calling `__getitem__` via `self[tuple(key)]`, is it possible to stick all of this logic into that function to avoid potentially creating and then throwing away a `DataArray` object?

Reply:
That makes way more sense than what I did here. I was able to move it into `__getitem__` but I needed to add some logic to check how the calculated array was being indexed but I think this is a better approach overall.